chatkit-core/src/test/kotlin/com/pusher/chatkit/rooms/RoomEventParserSpec.kt

```diff
@@ -0,0 +1,76 @@
+package com.pusher.chatkit.rooms
+
+import com.google.common.truth.Truth
+import com.pusher.chatkit.TestFileReader
+import com.pusher.chatkit.rooms.api.CreateRoomResponse
+import com.pusher.chatkit.rooms.api.JoinRoomResponse
+import com.pusher.chatkit.util.parseAs
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+
+object RoomEventParserSpec : Spek({
+    val testFileReader = TestFileReader("/json/rooms")
+
+    describe("when creating a room example from the docs") {
+
+        val result = testFileReader.readTestFile("room_created-docs.json")
+                .parseAs<CreateRoomResponse>().successOrThrow()
+
+        it("creates a valid room") {
+            val room = result.room
+
+            Truth.assertThat(room.id).isEqualTo("ac43dfef")
```

Comment:
as no we have simple ApiType without any init/mapping logic it would be better to create expected room and simply run one assertion:
```kotlin
assertThat(actualRoom).isEqualTo(expectedRoom)
```
the advantage is that if more than one bit gets broken the test clearly will report all the problems as opposed to other issues being hidden behind the first failure, also that expectedRoom can be reused for join room response below (like indeed it is reused in the docs)

Reply:
Have updated to your recommendation. 

I think that maybe we could have kept both? It would have be nice to check each field individually first so that if one does fail it's more helpful on which field failed - with just one check you have to pattern match the failure log yourself to spot which bit failed. We could then check the complete object matches at the end. If that makes sense?

Reply:
Thanks Danielle for addressing, it's much nicer now :)

I see the point about bigger message describing actual-expected discrepancy in the case of a failure. It is a bit of a cons but overfall I think it's the best trade-off. Alternative would be really to have a test per each property (what some folks do), however then it's a lot of tests, boiler-plate and repetition. Also objects shouldn't have too many properties. `Room` here has a bit more over what is usually recommended as max but still it's a reasonable number. A quick look at the failure message still would provide insight on what fields haven't been parsed correctly.

Having also additional corresponding single test with multiple assertions would create redundancy and still just the first assertion failure would be reported, hence I don't see it as a solution close to optimum.